CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+
+Schema Versioning Overview
+--------------------------
+
+Individual schemas may have their own internal [version](https://felis.lsst.io/user-guide/model.html#schema-version) for tracking changes, which is defined at the top level of the file, as in [apdb.yaml](yml/apdb.yaml).
+(This is distinct from tags or versions of sdm_schemas itself, and not all schemas may be using these internal versions.)
+This version may need to be updated when making changes.
+
+Usage of semantic versioning is recommended, which defines a version like `MAJOR.MINOR.PATCH`, such as `1.2.3`.
+
+Semantic versioning typically follows the following generic guidelines:
+
+- `MAJOR`: Breaking changes that are not backward compatible.
+- `MINOR`: Backward-compatible changes.
+- `PATCH`: Backward-compatible changes, typically those that do not change the database schema.
+
+These rules would typically not be followed strictly for schema changes. For instance, some operations which are technically backward compatible, such as adding a table, should likely result in a `MAJOR` version increment rather than `MINOR`, as they constitute an important and significant update which would usually require major updates to client queries and APIs.
+
+Versioning Guidelines
+---------------------
+
+Below are some general guidelines that can be used for incrementing the schema version based on what changes are being made:
+
+| Operation | Increment | Notes |
+|--------|-----------|-------|
+| Add table | MAJOR | Even if backward compatible, new tables often signal major functionality changes, usually requiring clients to update their code or queries. |
+| Remove table | MAJOR | Removing a table may break existing queries and client interfaces, making it incompatible with previous versions. |
+| Add column | MAJOR | Adding a column may require clients to adapt to the new data structure (e.g., parsing new fields), treated as a major change for consistency. |
+| Remove column | MAJOR | Removing a column breaks queries and code expecting that column, thus not backward compatible. |
+| Change column type| MAJOR | Changing a column’s data type can break clients expecting a different type, forcing a major version increment. |
+| Add index | PATCH | Indexes improve performance without changing the schema’s interface, making this a safe, backward-compatible improvement. |
+| Add constraint | MAJOR | New constraints can invalidate previously acceptable data, breaking clients that rely on the old lenient rules. |
+| Change constraint | MAJOR | Altered constraints can break clients relying on the previous constraint behavior, making this a non-backward-compatible change. |
+| Remove constraint | MINOR | Removing a constraint generally relaxes restrictions. It’s unlikely to break clients, but it changes behavior enough to warrant more than a patch. |
```

Comment:
Constraint changes are not at all clear case. These only affect database itself, client code usually does not care about constraint definitions. I guess you may be thinking in the terms of semver which is not quite applicable to these cases. It may be better to say that it should be decided on a case-by-case basis.

Reply:
Removed this section entirely.